website_sharing_whole_class_mosaic_plot_translation.Rmd

```diff
@@ -0,0 +1,507 @@
+### 3.7 Divider function。分割类型。
+每个部分有四个选项。
+
+* vspine：宽度不变，高度可变。
+* hspine：高度不变，宽度可变。
+* vbar：高度不变，宽度不同。
+* hbar：宽度不变，高度不同。
+
+```{r}
+hbar <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq), divider="hbar", na.rm=TRUE) + labs(x=" ", title='divider = "hbar"') 
+
+
+hspine <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq),  divider="hspine", na.rm=TRUE) + labs(x=" ", title='divider = "hspine"') 
+
+
+vbar <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq), divider="vbar", na.rm=TRUE) + labs(y=" ", x="", title='divider = "vbar"') 
+ 
+vspine <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq), divider="vspine", na.rm=TRUE) + labs(y=" ", x="", title='divider = "vspine"') 
+
+hbar
+hspine 
+vbar
+vspine
+```
+
+### 3.8 使用一个或多个变量进行分区
+
+* mosaic()
+ + 默认
+ + 将在交替的方向使用spine
+ + 从横向的spine开始
+* mosaic("v")
+ + 从竖向spine开始，然后交替进行。
+* ddecker()
+ + 选择n-1个水平spine，并以竖向spine结束
+* 定义每种类型的分区
+ + c("hspine"、"vspine"、"hbar")
+ 
+```{r}
+h_mosaic <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=mosaic("h")) +
+  theme(axis.text.x=element_blank(), legend.position="none") + 
+  labs(x=" ", title='divider= mosaic()')
+
+v_mosaic <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=mosaic("v")) +
+  theme(axis.text.x=element_blank()) + 
+  labs(x=" ", title='divider= mosaic("v")')
+
+
+doubledecker <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=ddecker()) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= ddecker()')
+
+h_mosaic
+v_mosaic
+doubledecker
+
+```
+ 
+
+```{r}
+mosaic4 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("vspine", "vspine", "hbar")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("vspine", "vspine", "hbar")') 
+
+mosaic5 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("hbar", "vspine", "hbar")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("hbar", "vspine", "hbar")')
+
+mosaic6 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("hspine", "hspine", "hspine")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("hspine", "hspine", "hspine")')
+
+mosaic7 <-  ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq, Gender, Region), fill=FlightFreq), na.rm=T, divider=c("vspine", "vspine", "vspine")) +
+  theme(axis.text.x=element_blank()) +
+  labs(x=" ", title='divider= c("vspine", "vspine", "vspine")')
+
+mosaic4
+mosaic5
+mosaic6
+mosaic7
+```
+
+
+### 3.9 geom_mosaic: offset
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.